docs/developing.md

```diff
@@ -54,6 +54,45 @@ Read more about what parameters you can pass [here][catch2].
 
 [catch2]: https://github.com/catchorg/Catch2/blob/v2.5.0/docs/command-line.md#specifying-which-tests-to-run
 
+### Value Generator Tests
+
+When modifing or creating new value generators it is highly valuable and important to test the implementation with a test case and to provide an example of the generator usage.
+
+#### Authroing Test Cases
+
+The test cases are implemented in src/value_generators/test/DocumentGeneratorTestCases.yml using the following template
+
+```
+ - Name: 
+    GivenTemplate:
+      <template>
+    ThenReturns:
+    - <expected output 1>
+    - <expected output 2>
+    - <expected output 3>
+```
+
+   Note:
+
+   *  The tests are executed in iterations, one for each of the expected outpus. 
+   For each iteration the `GivenTemplate` is compiled and evaulated with the declared
+   value generators in it. Then the actual and expected outputs are compared.
+   * In the sample above the template will be evaluated 3 times and the test will run in 3 iterations.
+   * Random generators declared in the templates are created only once and reused for all iterations. The produced values are incremental.
```

Comment:
Do you want to say something about the data being random by reproducible. So when adding a test you may need to run the test first to get results, verify they are reasonable results, and then put them into the expected value?